bigtable/admin/table_admin_test.cc

```diff
@@ -161,3 +165,108 @@ TEST_F(TableAdminTest, ListTablesUnrecoverableFailures) {
   // After all the setup, make the actual call we want to test.
   EXPECT_THROW(tested.ListTables(btproto::Table::FULL), std::exception);
 }
+
+/// @test Verify that bigtable::TableAdmin::Create works in the easy case.
+TEST_F(TableAdminTest, CreateTableSimple) {
+  using namespace ::testing;
+  using namespace bigtable::chrono_literals;
+
+  bigtable::TableAdmin tested(client_, "the-instance");
+  auto mock_create_table = [](grpc::ClientContext* ctx,
+                              btproto::CreateTableRequest const& request,
+                              btproto::Table* response) {
+    EXPECT_EQ("projects/the-project/instances/the-instance", request.parent());
+    std::string expected_text = R"""(
+parent: 'projects/the-project/instances/the-instance'
+table_id: 'new-table'
+table {
+  column_families {
+    key: 'f1'
+    value { gc_rule { max_num_versions: 1 }}
+  }
+  column_families {
+    key: 'f2'
+    value { gc_rule { max_age { seconds: 1 }}}
+  }
+  granularity: MILLIS
+}
+initial_splits { key: 'a' }
+initial_splits { key: 'c' }
+initial_splits { key: 'p' }
+)""";
+    btproto::CreateTableRequest expected;
+    EXPECT_TRUE(google::protobuf::TextFormat::ParseFromString(expected_text,
+                                                              &expected));
+    std::string delta;
+    google::protobuf::util::MessageDifferencer differencer;
+    differencer.ReportDifferencesToString(&delta);
+    EXPECT_TRUE(differencer.Compare(expected, request)) << delta;
```

Comment:
Is this the only thing we have for protobuf comparisons? There's no `ASSERT_THAT` gMock matcher support?

I couldn't find it; just asking if you know of anything that's more readable than `MessageDifferencer`.

Reply:
I did look too, could not find anything in `googletest`, nor in `protobuf`.  At least the output is readable with the `MessageDifferencer`, though yes, the test code is less than awesome.